sklearn/model_selection/_search.py

```diff
@@ -552,17 +565,39 @@ def _fit(self, X, y, labels, parameter_iterable):
 
         # Computed the (weighted) mean and std for all the candidates
         weights = test_sample_counts if self.iid else None
-        means = np.average(test_scores, axis=1, weights=weights)
-        stds = np.sqrt(np.average((test_scores - means[:, np.newaxis]) ** 2,
-                                  axis=1, weights=weights))
+
+        time = np.array(_, dtype=np.float64).reshape(n_candidates, n_splits)
+        time_means = np.average(time, axis=1, weights=weights)
+        time_stds =  np.sqrt(
+                np.average((time - time_means[:, np.newaxis]) ** 2,
+                           axis=1, weights=weights))
+        if self.return_train_score:
+            train_means = np.average(train_scores, axis=1, weights=weights)
```

Comment:
It doesn't make sense to weight the train scores by the number of test samples. Perhaps we should weight by the number of training samples when `iid=True`, or report an unweighted average. For times, weighting by training instances is more appropriate than by test, but leaving it unweighted might be best.